src/libostree/ostree-repo-pull.c

```diff
@@ -4172,6 +4152,42 @@ ostree_repo_pull_with_options (OstreeRepo             *self,
                                  g_strdup (delta),
                                  csum_data);
           }
+
+        if (g_variant_lookup (additional_metadata, OSTREE_SUMMARY_MODE, "s", &remote_mode_str) &&
+            g_variant_lookup (additional_metadata, OSTREE_SUMMARY_TOMBSTONE_COMMITS, "b", &tombstone_commits))
+          {
+            if (!ostree_repo_mode_from_string (remote_mode_str, &pull_data->remote_mode, error))
+              goto out;
+            pull_data->has_tombstone_commits = tombstone_commits;
+          }
+        else if (pull_data->remote_repo_local == NULL)
+          {
+            /* Fall-back path which loads the necessary config from the remote’s
+             * `config` file. Doing so is deprecated since it means an
+             * additional round trip to the remote for each pull. No need to do
+             * it for local pulls. */
+            if (!load_remote_repo_config (pull_data, &remote_config, cancellable, error))
+              goto out;
+
+            if (!ot_keyfile_get_value_with_default (remote_config, "core", "mode", "bare",
+                                                    &remote_mode_str, error))
+              goto out;
+
+            if (!ostree_repo_mode_from_string (remote_mode_str, &pull_data->remote_mode, error))
+              goto out;
+
+            if (!ot_keyfile_get_boolean_with_default (remote_config, "core", "tombstone-commits", FALSE,
+                                                      &pull_data->has_tombstone_commits, error))
+              goto out;
+
+            if (pull_data->remote_mode != OSTREE_REPO_MODE_ARCHIVE)
+              {
+                g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                             "Can't pull from archives with mode \"%s\"",
+                             remote_mode_str);
+                goto out;
+              }
```

Comment:
This check should be done in both the new and old paths, no?

Reply:
Actually, no. Turns out it should only be done on the `pull_data->remote_repo_local == NULL` path (the old behaviour) or a lot of the unit tests fail. I don’t know why this is. `load_remote_repo_config()` was previously only called on that path.

Reply:
Ahh yes, you're right.

> Turns out it should only be done on the pull_data->remote_repo_local == NULL path (the old behaviour) or a lot of the unit tests fail. I don’t know why this is.

Probably because we support `pull-local` from repos that aren't archive repos.